charger/easee.go

```diff
@@ -301,6 +307,9 @@ func (c *Easee) Enable(enable bool) error {
 	uri := fmt.Sprintf("%s/chargers/%s/commands/%s", easee.API, c.charger, action)
 	_, err := c.Post(uri, request.JSONContent, nil)
 
+	// pause/resume will async adjust charger current, wait for this to happen
+	time.Sleep(5 * time.Second)
```

Comment:
das ist hässlich und konnte bisher vermieden werden. Warum müssen wir hier warten? Workaround möglich?

Reply:
Pause und Resume ändern den Modus des Chargers. Zeitversetzt setzen sie die dynamicChargerCurrent auf 0 bzw MaxCurrent.  Nach dem Resume Call setzen wir ja die Current (bspw. auf die gespeicherten 6A). Kurz danach überschreibt Resume unseren Wert durch MaxCurrent (32A).

Konkret macht es beim ´resume´ Ärger. Wir könnten dieses Statement ins drauffolgende If packen.

Reply:
Wenn wir das wissen könnten wir ja auch kurzzeitig einen fiktiven Wert in die `dyn...Current` schreiben?

Reply:
Was meinst du mit fiktivem Wert? Wir schreiben ja einen Wert (den letzten gespeicherten). Der wird nur wieder überschrieben.

Reply:
Ich meine

```
lock()
dynamicChargerCurrent = 6 // Fantasiezahl > 0
unlock()
```

damit anschließend der Vergleich "enabled" sagt. Annahme: wird nicht wieder auf Null aktualisiert solange am Charger nix umgeschaltet wird.

Reply:
Es geht mir hier gar nicht um enabled. Es geht darum, dass der Charger bspw. nach nach aktivieren auf Max-Current läuft.

Konkretes Scenario:

- Modus Off
- Nutzer > Modus Min+PV
- > API: Resume
- > API: Settings (6A)
- < STATUS: dynamicChargerCurrent  6A (getriggert durch settings)
- < STATUS: dynamicChargerCurrent 32A (getriggert durch resume)

Ladevorgang läuft fälschlicherweise auf Max-Current und nicht auf 6A.

Reply:
Wie wärs denn wenn wir das hier gar nicht handlen, sondern im SignalR Listener immer wenn dynamicCurrent != gesetzter Strom ein Update triggern? Dann ist uns egal wie lange der Zustand anhält. 

Im Prinzip ist das ein doofer Bug und für ein Lastmanagement nicht vorhersehbar, umso näher sollten wir an den Zeitpunkt ran wo es passiert.

Reply:
Habe es jetzt erfolgreich mit 1s Delay getestet und nur auf Resume beschränkt.

Reply:
> Wie wärs denn wenn wir das hier gar nicht handlen, sondern im SignalR Listener immer wenn dynamicCurrent != gesetzter Strom ein Update triggern? Dann ist uns egal wie lange der Zustand anhält.

Ich kann das mal ausprobieren. Ich seh da aber ein Risiko für Loops. Bin mir nicht sicher dass das ne schönere Lösung ist.

Reply:
@andig schau noch mal drauf.  Habs jetzt wie vorgeschlagen und ohne Sleep umgesetzt. Die Reihenfolge und das Timing funktioniert.